src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -200,18 +206,22 @@ default ClassDesc arrayType(int rank) {
         if (rank <= 0) {
             throw new IllegalArgumentException("rank " + rank + " is not a positive value");
         }
+        String desc = descriptorString();
         try {
-            int currentDepth = ConstantUtils.arrayDepth(descriptorString());
+            int currentDepth = arrayDepth(desc);
             netRank = Math.addExact(currentDepth, rank);
-            if (netRank > ConstantUtils.MAX_ARRAY_TYPE_DESC_DIMENSIONS) {
+            if (netRank > MAX_ARRAY_TYPE_DESC_DIMENSIONS) {
                 throw new IllegalArgumentException("rank: " + netRank +
                                                    " exceeds maximum supported dimension of " +
-                                                   ConstantUtils.MAX_ARRAY_TYPE_DESC_DIMENSIONS);
+                                                   MAX_ARRAY_TYPE_DESC_DIMENSIONS);
             }
         } catch (ArithmeticException ae) {
             throw new IllegalArgumentException("Integer overflow in rank computation");
         }
-        return ClassDesc.ofDescriptor("[".repeat(rank) + descriptorString());
+        if (desc.length() == 1 && desc.charAt(0) == 'V') {
+            throw new IllegalArgumentException(String.format("not a valid reference type descriptor: %sV", "[".repeat(rank)));
```

Comment:
```suggestion
            throw new IllegalArgumentException(String.format("not a valid reference type descriptor: %sV", "[".repeat(netRank)));
```
Or should we override this in `PrimitiveClassDescImpl`, which can bypass the rank sum computation?

Reply:
`currentDepth` must be 0 in this case, so `rank` or `netRank` doesn't matter. Overriding in `PrimitiveClassDescImpl` sounds reasonable, but then perhaps default method should be removed, too, since it would look strange to have the default method be specialized for instance/array types. Sounds like a CSR might be needed(?), so let's do that in a follow up.

Reply:
I don’t think a **CSR** is needed for changing the `default`ness of JDK‑sealed interface methods, as source and binary compatibility for external users is unaffected.

Reply:
A CSR is required when defaultness is removed, as in https://bugs.openjdk.org/browse/JDK-8309755